code/__HELPERS/icon_smoothing.dm

```diff
@@ -41,18 +41,61 @@
 		if(!AM.anchored)
 			return 0
 
-		for(var/direction in alldirs)
+		for(var/direction in cardinal)
 			AM = find_type_in_direction(A, direction)
-			if(istype(AM))
-				if(AM.anchored)
-					adjacencies |= 1 << direction
-			else
-				if(AM)
-					adjacencies |= 1 << direction
+			if(istype(AM) && AM.anchored)
+				adjacencies |= 1 << direction
+			else if(AM)
+				adjacencies |= 1 << direction
```

Comment:
Nope, because the first check is to know if I have a movable or not, it fails if it isn't movable and not null, so I have to check if it's null in the else.